compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -1763,8 +1764,6 @@ fn compare_const_predicate_entailment<'tcx>(
     let impl_ct_predicates = tcx.predicates_of(impl_ct.def_id);
     let trait_ct_predicates = tcx.predicates_of(trait_ct.def_id);
 
-    check_region_bounds_on_impl_item(tcx, impl_ct, trait_ct, false)?;
```

Comment:
On lifetime param mismatches, we ICE'd in `tcx.type_of(trait_ct.def_id).instantiate(tcx, trait_to_impl_args)` a few lines further up for obvious reasons. That's on me, I messed that up in the initial GCI PR.